src/middleware/router.coffee

```diff
@@ -76,53 +73,58 @@ sendRequestToRoutes = (ctx, routes, next) ->
   if containsMultiplePrimaries routes
     return next new Error "Cannot route transaction: Channel contains multiple primary routes and only one primary is allowed"
 
-  for route in routes
-    path = getDestinationPath route, ctx.path
-    options =
-      hostname: route.host
-      port: route.port
-      path: path
-      method: ctx.request.method
-      headers: ctx.request.header
-      agent: false
-      rejectUnauthorized: true
-      key: tlsKey
-      cert: tlsCert
-      secureProtocol: 'TLSv1_method'
-
-    if ctx.request.querystring
-      options.path += '?' + ctx.request.querystring
-
-    if options.headers && options.headers.authorization
-      delete options.headers.authorization
-
-    if route.username and route.password
-      options.auth = route.username + ":" + route.password
-
-    if options.headers && options.headers.host
-      delete options.headers.host
-
-    if route.primary
-      promise = sendRequest(ctx, route, options)
-      .then (response) ->
-        if response.headers?['content-type']?.indexOf('application/json+openhim') > -1
-          # handle mediator reponse
-          responseObj = JSON.parse response.body
-          ctx.mediatorResponse = responseObj
-          # then set koa response from responseObj.response
-          setKoaResponse ctx, responseObj.response
-        else
-          setKoaResponse ctx, response
-      .fail (reason) ->
-        # on failure
-        handleServerError ctx, reason
-    else
-      promise = buildNonPrimarySendRequestPromise ctx, route, options, path
+  Keystore.findOne {}, (err, keystore) ->
```

Comment:
Actually I can't do this as I need the key and cert for the options of each request (https://github.com/jembi/openhim-core-js/pull/359/files#diff-ffc48656de2a6c6b5ee8a785b33109b0R88). I don't want to make this global as then if someone changes this in the UI it won't have immediate effect.